coremltools/models/utils.py

```diff
@@ -1675,3 +1681,394 @@ def randomize_weights(mlmodel: "_ct.models.MLModel"):
     )
 
     return randomized_mlmodel
+
+
+def bisect_model(
+    model_path: str,
+    output_dir: str,
+    remove_original: _Optional[bool] = False,
+    merge_chunks_to_pipeline: _Optional[bool] = False,
+    check_output_correctness: _Optional[bool] = False,
+):
+    """
+    Utility function to split a mlpackage model into two mlpackages of approximately same file size.
+
+    The implementation of this util is mainly migrated from the below source:
+    https://github.com/apple/ml-stable-diffusion/blob/main/python_coreml_stable_diffusion/chunk_mlprogram.py
+
+    Parameters
+    ----------
+    model_path: str
+        Path to the mlpackage file to be split into two mlpackages of approximately same file size.
+
+    output_dir: str
+        Path to output directory where the two model chunks / pipeline model would be saved.
+        If the `model_path` is `{path}/{model_name}.mlpackage`, the chunks models are going to be saved as:
+
+        1. first chunk model: `{output_dir}/{model_name}_chunk1.mlpackage`
+        2. second chunk model: `{output_dir}/{model_name}_chunk2.mlpackage`
+        3. chunked pipeline model: `{output_dir}/{model_name}_chunked_pipeline.mlpackage`
+
+    remove_original: bool
+        If True, removes the original (non-chunked) model to avoid duplicating storage.
+
+    merge_chunks_to_pipeline: bool
+        If True, model chunks are managed inside a single pipeline model for easier asset maintenance.
+
+    check_output_correctness: bool
```

Comment:
Maybe this should be True by default.

Reply:
Agree! I think we can handle this in the ml-stable-diffusion itself @aseemw

Reply:
I think it is better to make the default `False`,
given that when users call this bisect function,
they are usually already facing the memory issue, by running the original model :) ?

Reply:
Ah, yeah looking at the implementation, that would use a lot of memory. Perhaps, we could rework it to make it more efficient. So we're only loading one model at a time.

For example with chunk 1 model - we could save the results right after its convert. Then call `del model_chunk1` before we call `_gc.collect()`.

The implementation would be a bit more complex, but it would use a lot less memory at any one time. What do you think?

Reply:
`check_output_correctness ` is comparing the outputs from "the original model" with the "chunk models / pipeline model".
The issue is, when running "the original model", it could already produce OOM issue.
@aseemw is this the case why you added the script in the very beginning?

Reply:
Made it default to True